sdv/single_table/base.py

```diff
@@ -101,16 +101,16 @@ def _update_default_transformers(self):
             for sdtype, transformer in self._model_sdtype_transformers.items():
                 self._data_processor._update_transformers_by_sdtypes(sdtype, transformer)
 
-    def _check_original_metadata_updated(self):
-        if not hasattr(self, '_original_metadata'):
+    def _check_input_metadata_updated(self):
+        if not hasattr(self, '_input_metadata'):
             unified_metadata = Metadata.load_from_dict(self.metadata.to_dict())
-            setattr(self, '_original_metadata', unified_metadata)
+            setattr(self, '_input_metadata', unified_metadata)
 
-        if isinstance(self._original_metadata, Metadata):
-            metadata = self._original_metadata._convert_to_single_table()
+        if isinstance(self._input_metadata, Metadata):
+            metadata = self._input_metadata._convert_to_single_table()
 
         else:
-            metadata = self._original_metadata
+            metadata = self._input_metadata
```

Comment:
This approach doesn't resolve [this](https://github.com/sdv-dev/SDV/issues/2463) issue (if you run the code snippet the warning doesn't surface).

The reason is that it checks `if _input_metadata._updated` is True in line 115, but that may not be the case since only `medatada._updated` gets set to True by the user-facing methods.

Reply:
<img width="1159" alt="image" src="https://github.com/user-attachments/assets/f2ded4d5-f63d-4cf3-9c7a-62e3b46ce615" />
I'm not sure if I understand your concern here , but I think it works as intended.

Reply:
This gets fixed with your latest commit. My only concern would be if the user changes what metadata instance they are using - by saving and loading the metadata for example - but maybe we don't care about such cases?